packages/protocol/test/ZkAsset/ZkAssetMintable.js

```diff
@@ -137,6 +152,208 @@ contract('ZkAssetMintable', (accounts) => {
             expect(initialBalance).to.equal(0);
             expect(finalBalance).to.equal(kPublic * scalingFactor);
         });
+
+
+        it('should perform mint using confidentialTransferFrom()', async () => {
+            const [owner, recipient1, recipient2] = aztecAccounts;
+            const inputNoteOwners = [recipient1, recipient2];
+
+            // Create dummy ZkAsset, through which confidentialTransferFrom() will be called
+            const zkAssetMintableTest = await ZkAssetMintableTest.new();
+            await zkAssetMintableTest.setZkAssetMintableAddress(zkAssetMintable.address);
+
+            const erc20TotalSupply = (await erc20.totalSupply()).toNumber();
+            const initialAceBalance = (await erc20.balanceOf(ace.address)).toNumber();
+            const initialRecipientBalance = (await erc20.balanceOf(accounts[1])).toNumber();
+            expect(initialAceBalance).to.equal(0);
+            expect(erc20TotalSupply).to.equal(0);
+            expect(initialRecipientBalance).to.equal(0);
+
+
+            const newTotalMinted = note
+                .create(owner.publicKey, 50);
+            const oldTotalMinted = note
+                .createZeroValueNote();
+
+            const mintedNotes = [
+                note.create(recipient1.publicKey, 20),
+                note.create(recipient2.publicKey, 30),
+            ];
+
+            const mintProof = proof.mint
+                .encodeMintTransaction({
+                    newTotalMinted, // 50
+                    oldTotalMinted, // 0
+                    adjustedNotes: mintedNotes, // 30 + 20
+                    senderAddress: zkAssetMintable.address,
+                });
+
+            const { receipt: mintReceipt } = await zkAssetMintable
+                .confidentialMint(MINT_PROOF, mintProof.proofData);
+
+            expect(mintReceipt.status).to.equal(true);
+            const erc20TotalSupplyAfterMint = (await erc20.totalSupply()).toNumber();
+            expect(erc20TotalSupplyAfterMint).to.equal(0);
+
+            // call confidentialApprove()
+            const indexes = [0, 1];
+            await Promise.all(indexes.map((i) => {
+                const { signature } = signNote(
+                    zkAssetMintable.address,
+                    mintedNotes[i].noteHash,
+                    zkAssetMintableTest.address,
+                    inputNoteOwners[i].privateKey
+                );
+                const concatenatedSignature = signature[0] + signature[1].slice(2) + signature[2].slice(2);
```

Comment:
This is... ugly. I believe this got changed a lot on `develop` and cleaned away so I think all good.

Reply:
I agree. It did get changed to an extent on `develop`, there is a function defined at the top of the `ZkAssetOwnable.js` script: 

    const confidentialApprove = async (indexes, notes, aztecAccounts) => {
        await Promise.all(
            indexes.map((i) => {
                const signature = signer.signNote(
                    zkAssetOwnable.address,
                    notes[i].noteHash,
                    zkAssetOwnableTest.address,
                    aztecAccounts[i].privateKey,
                );
                // eslint-disable-next-line no-await-in-loop
                return zkAssetOwnable.confidentialApprove(notes[i].noteHash, zkAssetOwnableTest.address, true, signature);
            }),
        );
    };

It's a bit nicer, but there's still scope to improve - the function could be moved to a `utils` module. Will leave this as is on `master` for now, make a change on `develop` in my current PR.